src/goto-symex/goto_symex_state.cpp

```diff
@@ -141,7 +149,7 @@ bool goto_symex_statet::constant_propagation(const expr2tc &expr) const
     bool noconst = true;
 
     expr->foreach_operand([this, &noconst](const expr2tc &e) {
-      if (noconst && !constant_propagation(e))
+      if (noconst && (!constant_propagation(e) || is_with2t(e)))
```

Comment:
is this needed?

Reply:
Yes, It is necessary and without it there will be serious performance issues.

Reply:
Then at least add a comment explaining why this is here. Can you share an example where this is required?

Reply:
Use the comment to describe why this is needed. It is obvious that we skip `with2t`s but why are we doing it?